internal/pkg/template/templates/environment/partials/cdn-resources.yml

```diff
@@ -17,8 +17,40 @@ CloudFrontDistribution:
             OriginProtocolPolicy: match-viewer # Will need to change with TLS termination and Alias usage
           DomainName: !GetAtt PublicLoadBalancer.DNSName
           Id: !Sub 'copilot-${AppName}-${EnvironmentName}-origin'
-      # ViewerCertificate: # Used for Aliases
-      #   AcmCertificateArn: # Imported certificate goes here
-      #   CloudFrontDefaultCertificate: true # If importing certificate, set to false
-      #   MinimumProtocolVersion: TLSv1 # Default for now, may need to change later
-      #   SsLSupportMethod: sni-only # Changing this can incur costs to the user
+{{- if not .PublicHTTPConfig.ImportedCertARNs}}
+      ViewerCertificate:
+        !If
+          - DelegateDNS
+          - AcmCertificateArn: !Ref CertificateReplicator
+            MinimumProtocolVersion: TLSv1
+            SslSupportMethod:  sni-only
+          - !Ref AWS::NoValue
+
+CertificateReplicatorFunction:
+  Type: AWS::Lambda::Function
+  Condition: DelegateDNS
+  Properties:
+    {{- with $cr := index .CustomResources "CertificateReplicatorFunction" }}
+    Code:
+      S3Bucket: {{$cr.Bucket}}
+      S3Key: {{$cr.Key}}
+    {{- end}}
+    Handler: "index.certificateReplicateHandler"
+    Timeout: 900
+    MemorySize: 512
+    Role: !GetAtt 'CustomResourceRole.Arn'
+    Runtime: nodejs12.x
+    
+CertificateReplicator:
```

Comment:
nit
```suggestion
ReplicatedCertificate:
```

Reply:
Technically this is still a custom resource. It is just we set the physical resource ARN to be the same as the replicated cert. I'd still prefer to use this name also to align with the other custom resource action.